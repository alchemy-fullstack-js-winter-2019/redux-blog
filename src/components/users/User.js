import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function User({ user }) {
  return (
    <>
      <Link to={`/users/${user.id}`} key={user.id}><h2>{user.name}</h2></Link>
      <p>{user.username}</p>
      <p>{user.email}</p>
    </>
  );
}

User.propTypes = {
  user: PropTypes.object.isRequired
};
