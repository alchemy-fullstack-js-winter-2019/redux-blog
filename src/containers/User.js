import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import UserDetail from '../components/users/UserDetail';
import { getUser } from '../selectors/users/users';
import { fetchUser } from '../action/users';
import { withRouter } from 'react-router-dom';

class User extends PureComponent {
  static propTypes = {
    user: PropTypes.object.isRequired,
    fetch: PropTypes.func.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string.isRequired
      })
    })
  };

  componentDidMount() {
    this.props.fetch(this.props.match.params.id);
  }

  render() {
    return (
      <>
      <UserDetail user={this.props.user}/>
      </>
    );
  }
}

const mapStateToProps = state => ({
  user: getUser(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(fetchUser());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(User));
