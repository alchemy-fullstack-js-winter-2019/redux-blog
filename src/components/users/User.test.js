import React from 'react';
import renderer from 'react-test-renderer';
import User from './User';
import { BrowserRouter as Router } from 'react-router-dom';

describe('User', () => {
  it('matches a snapshot', () => {
    const user = {
      name: 'name',
      username: 'username',
      email: 'email@email'
    };
    const tree = renderer.create(
      <Router>
        <User user={user}/>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
