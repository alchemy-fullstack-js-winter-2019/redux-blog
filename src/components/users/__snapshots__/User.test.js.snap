// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a snapshot 1`] = `
Array [
  <a
    href="/users/undefined"
    onClick={[Function]}
  >
    <h2>
      name
    </h2>
  </a>,
  <p>
    username
  </p>,
  <p>
    email@email
  </p>,
]
`;
